opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -564,6 +570,10 @@ namespace {
                 state.saturation[pu.phase_pos[ Oil ]] = so;
             }
         }
+
+        if( nextvar == int(vars.size()) )
+            return state;
+
         // Qs.
```

Comment:
I think this is a bit confusing. It is correct, but I'd much rather have:
```
if (wellsActive()) {
    // Assign state.qs and state.bhp as before
    ...
} else {
    state.qs = ADB::null();
    state.bhp = ADB::null();
}
```
That would leave the function with a single return statement, which is preferable, also it is better to have the explicit `if (wellsActive())` than the obscure `if (nextvar == vars.size())`.